src/App.tsx

```diff
@@ -15,35 +16,116 @@
+        sorted.sort((a, b) => {
+          setAlphabetically(!alphabeticallyPressed);
+          if (byLengthPressed) {
+            setByLength(!byLengthPressed);
+          }
+          if (reversePressed) {
+            return a.localeCompare(b) * -1;
+          } else {
+            return a.localeCompare(b)
+          }
+        })
+        break;
+
+      case 'byLength':
+        sorted.sort((a, b) => {
+          setByLength(!byLengthPressed);
+          if (alphabeticallyPressed) {
+            setAlphabetically(!alphabeticallyPressed);
+          }
+          if (reversePressed) {
+            return (a.length - b.length) * -1;
+          } else {
+            return a.length - b.length
+          }
+        })
+        break;
+
+      case 'reverse':
+        sorted.reverse();
+        setReversePressed(!reversePressed);
+        break;
+
+      default:
+        setAlphabetically(false)
+        setByLength(false)
+        setReversePressed(false)
+        sorted = [...goodsFromServer];
+    }
+
+    setSortedGoods(sorted);
+  };
+
+  const renderResetButton = () => {
+    return (
+      <button type="button" 
+      className="button is-danger is-light"
+      onClick={() => {
+        sortGoods('');
+      }}
+      >
+        Reset
+      </button>
+    )
+   }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button type="button" 
+        className={classNames("button is-info", {"is-light": alphabeticallyPressed === false})}
+        onClick={() => {
+          sortGoods('alphabetically');
+        }}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button type="button" 
+        className={classNames("button is-success", {"is-light": byLengthPressed === false})}
+        onClick={() => {
+          sortGoods('byLength');
+        }}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button type="button" 
+        className={classNames("button is-warning", {"is-light": reversePressed === false})}
+        onClick={() => {
+          sortGoods('reverse');
+        }}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {
+        JSON.stringify(sortedGoods) !== JSON.stringify(goodsFromServer)
+          ? renderResetButton()
+          : null}
+
+       
       </div>
 
       <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {sortedGoods.map(good =>  (<li data-cy="Good" key = {good}>{good}</li>))}
```

Comment:
The `key` attribute for list items should be unique. Using the `good` value as a key is acceptable here since the goods are unique, but ensure that this remains true if the data changes.